java/src/main/java/com/logicalclocks/hsfs/engine/Utils.java

```diff
@@ -65,12 +68,18 @@
     int index = 0;
     for (StructField structField : dataset.schema().fields()) {
       // TODO(Fabio): unit test this one for complext types
-      features.add(new TrainingDatasetFeature(structField.name(), structField.dataType().catalogString(), index++));
+      features.add(new TrainingDatasetFeature(
+          structField.name().toLowerCase(), structField.dataType().catalogString(), index++));
     }
 
     return features;
   }
 
+  public Dataset<Row> sanetizeFeatureNames(Dataset<Row> dataset) {
+    return dataset.select(Arrays.asList(dataset.columns()).stream().map(f -> col(f).alias(f.toLowerCase())).toArray(
+        Column[]::new));
```

Comment:
You can actually just leave them strings instead of converting them to column, select should work just fine

Reply:
I know it looks stupid, but because of method overloading there is no select which takes just a single `String[]`. Instead the signature is `public Dataset<Row> select(String col, String... cols)` so I would have had to split the array first. I thought this is cleaner.